ui/app/models/azure/config.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import fieldToAttrs, { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+
+// Note: while the API docs indicate subscriptionId and tenantId are required, the UI does not enforce this because the user may pass these values in as environment variables.
+// https://developer.hashicorp.com/vault/api-docs/secret/azure#configure-access
+export default class AzureConfig extends Model {
+  @attr('string') backend; // dynamic path of secret -- set on response from value passed to queryRecord
+  @attr('string', {
+    label: 'Subscription ID',
+  })
+  subscriptionId;
+  @attr('string', {
+    label: 'Tenant ID',
+  })
+  tenantId;
+  @attr('string', {
+    label: 'Client ID',
+  })
+  clientId;
+  @attr('string', { sensitive: true }) clientSecret; // obfuscated, never returned by API
+  @attr('string', {
+    subText:
+      'The audience claim value for plugin identity tokens. Must match an allowed audience configured for the target IAM OIDC identity provider.',
+  })
+  identityTokenAudience;
+  @attr({
+    label: 'Identity token TTL',
+    helperTextDisabled:
+      'The TTL of generated tokens. Defaults to 1 hour, turn on the toggle to specify a different value.',
+    helperTextEnabled: 'The TTL of generated tokens.',
+    editType: 'ttl',
+  })
+  identityTokenTtl;
+  @attr('string') environment;
+  @attr({
+    label: 'Root password TTL',
+    editType: 'ttl',
+    helperTextDisabled:
+      'Specifies how long the root password is valid for in Azure when rotate-root generates a new client secret. Defaults to 182 days or 6 months, 1 day and 13 hours.',
+  })
+  rootPasswordTtl;
+
+  // for configuration details view
+  // do not include clientSecret because it is never returned by the API
+  get attrs() {
+    const keys = [
+      'subscriptionId',
+      'tenantId',
+      'clientId',
+      'identityTokenAudience',
+      'identityTokenTtl',
+      'environment',
+      'rootPasswordTtl',
+    ];
+    return expandAttributeMeta(this, keys);
+  }
```

Comment:
can we reuse the `formFields` getter here since `expandAttributeMeta` has already been called there? Another benefit is we only have to update one array of keys in the future if/when new attrs are added.
(I also think renaming `attrs` to be more specific is helpful here since `attrs` is used all over the place, internally and also historically with ember data models. 

I can't remember the struct exactly, but something like:
(In the future this could be extended to be an array of "hiddenParams" but for now we can just filter for clientSecret)
```js
  // for displaying configuration details
  // do not include clientSecret because it is never returned by the API
  get displayAttrs() {
   this.formFields.filter( attr => attr.name !== 'clientSecret')
  }
```